src/CommonITILValidation.php

```diff
@@ -770,8 +804,34 @@ public static function getNumberToValidate($users_id)
     final public static function getTargetCriteriaForUser(int $users_id, bool $search_in_groups = true): array
     {
         $target_criteria = [
-            static::getTableField('itemtype_target') => User::class,
-            static::getTableField('items_id_target') => $users_id,
+            'OR' => [
+                [
+                    static::getTableField('itemtype_target') => User::class,
+                    static::getTableField('items_id_target') => $users_id,
+                ],
+                'AND' => [
+                    [
+                        'OR' => [
+                            [
+                                User::getTableField('substitution_start_date')  => null,
+                            ],
+                            [
+                                User::getTableField('substitution_start_date')  => ['<=', $_SESSION['glpi_currenttime']],
+                            ],
+                        ],
+                    ], [
+                        'OR' => [
+                            [
+                                User::getTableField('substitution_end_date') => null,
+                            ],
+                            [
+                                User::getTableField('substitution_end_date') => ['>=', $_SESSION['glpi_currenttime']],
+                            ],
+                        ],
+                    ],
+                    ValidatorSubstitute::getTableField('users_id_substitute')  => $users_id,
```

Comment:
`TicketValidation::getTargetCriteriaForUser()` is used in `Glpi\Dashboard\Provider::nbTicketsGeneric()`, and `ValidatorSubstitute` is not joined there, so query will probaly fail.

There is no need to add joined tables for groups criteria due to usage of subqueries. I guess it is possible to do the same for users criteria.

Reply:
Shoud be partially solved with bc2048892d  . Check it carefully as I have a doubt with the subquery, but could not yet see it fail to produce the expected results.